tidoop-mr-lib-api/src/server.js

```diff
@@ -23,36 +23,134 @@
         reply(response);
-    }
+    } // handler
+});
+
+server.route({
+    method: 'GET',
+    path: '/tidoop/v1/jobs/',
+    handler: function(request, reply) {
+        console.log('Request: GET /tidoop/v1/jobs/');
+        throw new Error('Unsupported operation');
+    } // handler
+});
+
+server.route({
+    method: 'POST',
+    path: '/tidoop/v1/jobs/',
+    handler: function (request, reply) {
+        console.log('Request: POST /tidoop/v1/jobs/ ' + JSON.stringify(request.payload));
+
+        // check the request parameters
+        // TBD
+
+        // get the jobType
+        var jobType = request.payload.job_type;
+
+        // create a jobId
+        var jobId = 'tidoop_job_' + Date.now();
+
+        // create a new job entry in the database
+        tidoopMysql.addNewJob(jobId, jobType, function(error, result) {
+            if (error) {
+                console.log('The new job could not be added to the database. Details: ' + err);
+            } else {
+                // run the Filter MR job; the callback function will receive the complete output once it finishes
+                run(jobId,
+                    'hadoop',
+                    ['jar', tidoopMRLibPath,
+                    serverUtils.getMRJobByType(jobType),
+                    '-libjars', tidoopMRLibPath].concat(
+                        serverUtils.getParamsForMRJob(jobType, request.payload)),
+                    function(result) {
+                        console.log(result);
+                    }
+                );
+
+                // create the response
+                var response = '{job_id: ' + jobId + '}';
+                console.log("Response: " + response);
+
+                // return the response
+                reply(response);
+            }
+        });
+    } // handler
+});
+
+server.route({
+    method: 'GET',
+    path: '/tidoop/v1/jobs/{jobId?}',
+    handler: function (request, reply) {
+        console.log('Request: GET /tidoop/v1/jobs/' + request.params.jobId + '/');
+
+        // check the request parameters
+        // TBD
+
+        // get the jobId
+        var jobId = request.params.jobId;
+
+        // get the job status
+        var result = tidoopMysql.getJob(jobId, function (error, result) {
+            if (error) {
+                throw error;
+            } else {
+                // create the response
+                var response = '{job_id: ' + jobId + ', job_type: ' + result[0].jobType + ', start_time: ' +
+                    result[0].startTime + ', end_time: ' + result[0].endTime + ', map_progress: ' +
+                    result[0].mapProgress + ', reduce_progress: ' + result[0].reduceProgress + '}';
+                console.log("Response: " + response);
+
+                // return the response
+                reply(response);
+            } // if else
+        });
+    } // handler
+});
+
+server.route({
+    method: 'DELETE',
+    path: '/tidoop/v1/jobs/{jobId?}',
```

Comment:
Agreed on my side, although to avoid misconceptions remember to return a list of resources (as read in the text: "This resource allows you to view a list of questions"), although in same cases, such as when an unique id is passed, that list only includes 1 element :) I suggested to use the singular form since I only see the option to pass a jobId and I though you only planned to return the single job data ;) This is the use of the singular or plural form should be aligned with the returned data ;)